app/navigation/Navigation.scala

```diff
@@ -21,29 +21,48 @@ import models.pages.Page
 import models.pages._
 import play.api.mvc.Call
 import controllers.routes
+import helpers.DateTimeHelper
+import utils.Logger.logger
+import utils.SessionKeys
 
+import java.time.LocalDateTime
 import javax.inject.Inject
 
-class Navigation @Inject()() {
-  lazy val checkingRoutes: Map[Page, Option[String] => Call] = Map(
-    HasCrimeBeenReportedPage -> (_ => routes.CheckYourAnswersController.onPageLoad()),
-    WhenDidCrimeHappenPage -> (_ => routes.CheckYourAnswersController.onPageLoad())
+class Navigation @Inject()(dateTimeHelper: DateTimeHelper) {
+  lazy val checkingRoutes: Map[Page, (Option[String], UserRequest[_]) => Call] = Map(
+    HasCrimeBeenReportedPage -> ((_, request) => routeToMakingALateAppealOrCYAPage(request)),
+    WhenDidCrimeHappenPage -> ((_, request) => routeToMakingALateAppealOrCYAPage(request))
   )
 
-  lazy val normalRoutes: Map[Page, Option[String] => Call] = Map(
-    HasCrimeBeenReportedPage -> (_ => routes.CheckYourAnswersController.onPageLoad()),
-    WhenDidCrimeHappenPage -> (_ => routes.CrimeReasonController.onPageLoadForHasCrimeBeenReported(NormalMode))
+  lazy val normalRoutes: Map[Page, (Option[String], UserRequest[_]) => Call] = Map(
+    HasCrimeBeenReportedPage -> ((_, request) => routeToMakingALateAppealOrCYAPage(request)),
+    WhenDidCrimeHappenPage -> ((_, _) => routes.CrimeReasonController.onPageLoadForHasCrimeBeenReported(NormalMode))
   )
 
-  def nextPage(page: Page, mode: Mode, answer: Option[String] = None): Call = {
+  def nextPage(page: Page, mode: Mode, answer: Option[String] = None)(implicit userRequest: UserRequest[_]): Call = {
     mode match {
       case CheckMode => {
         //Added answer here so that we can add custom routing to pages that require extra data when answers change
-        checkingRoutes(page)(answer)
+        checkingRoutes(page)(answer, userRequest)
       }
       case NormalMode => {
-        normalRoutes(page)(answer)
+        normalRoutes(page)(answer, userRequest)
       }
     }
   }
+
+  def routeToMakingALateAppealOrCYAPage(userRequest: UserRequest[_]): Call = {
+    val dateSentParsed: LocalDateTime = LocalDateTime.parse(userRequest.session.get(SessionKeys.dateCommunicationSent).get)
+    //TODO: may need to call config to get the days
+    val daysResultingInLateAppeal: Int = 30
+    val dateTimeNow: LocalDateTime = dateTimeHelper.dateTimeNow
+    if(dateSentParsed.isBefore(dateTimeNow.minusDays(daysResultingInLateAppeal)) && userRequest.session.get(SessionKeys.lateAppealReason).isEmpty) {
```

Comment:
this is causing the late appeal page to be skipped if the user has already entered a reason and then goes back

https://user-images.githubusercontent.com/6122384/121500750-7fd85380-c9d6-11eb-9055-024be8217a40.mov

Reply:
fix implemented in https://github.com/hmrc/penalties-appeals-frontend/pull/30